src/easing.jl

```diff
@@ -0,0 +1,191 @@
+#=
+    # The four inputs t, b, c, d are defined as follows:
+
+    * t = current time (in any unit measure, but same unit as duration)
+    * b = starting value to interpolate
+    * c = the total change in value of b that needs to occur
+    * d = total time it should take to complete (duration)
+=#
+
+# Linear Easing functions
+EaseLinearNone(t::Real, b::Real, c::Real, d::Real) = c * t/d + b
+EaseLinearIn(t::Real, b::Real, c::Real, d::Real) = EaseLinearNone(t, b, c, d)
+EaseLinearOut(t::Real, b::Real, c::Real, d::Real) = EaseLinearNone(t, b, c, d)
+EaseLinearInOut(t::Real, b::Real, c::Real, d::Real) = EaseLinearNone(t, b, c, d)
+
+# Sine Easing functions
+EaseSineIn(t::Real, b::Real, c::Real, d::Real) = -c * cos(t/d * (π/2)) + c + b
+EaseSineOut(t::Real, b::Real, c::Real, d::Real) = c * sinf(t/d * (π/2)) + b
+EaseSineInOut(t::Real, b::Real, c::Real, d::Real) = -c/2 * (cos(π * t/d) - 1) + b
+
+# Circular Easing functions
+function EaseCircIn(t::Real, b::Real, c::Real, d::Real)
+    t /= d
+
+    return -c * (sqrt(1 - t^2) - 1) + b
+end
+function EaseCircOut(t::Real, b::Real, c::Real, d::Real)
+    t = t/d - 1
+
+    return c * sqrt(1 - t^2) + b
+end
+function EaseCircInOut(t::Real, b::Real, c::Real, d::Real)
+    if (t /= d/2) < 1
+        return -c/2 * (sqrt(1 - t^2) - 1) + b
+    end
+
+    t -= 2
+
+    return c/2 * (sqrt(1 - t^2) + 1) + b
+end
+
+# Cubic Easing functions
+function EaseCubicIn(t::Real, b::Real, c::Real, d::Real)
+    t /= d
+
+    return c * t^3 + b
+end
+function EaseCubicOut(t::Real, b::Real, c::Real, d::Real)
+    t = t/d - 1
+
+    return c * (t^3 + 1) + b
+end
+function EaseCubicInOut(t::Real, b::Real, c::Real, d::Real)
+    if (t /= d/2) < 1
+        return c/2 * t^3 + b
+    end
+
+    t -= 2
+
+    return c/2 * (t^3 + 2) + b
+end
+
+# Quadratic Easing functions
+function EaseQuadIn(t::Real, b::Real, c::Real, d::Real)
+    t /= d
+
+    return c * t^2 + b
+end
+function EaseQuadOut(t::Real, b::Real, c::Real, d::Real)
+    t /= d
+
+    return -c * t * (t - 2) + b
+end
+function EaseQuadInOut(t::Real, b::Real, c::Real, d::Real)
+    if (t /= d/2) < 1
+        return c/2 * t^2 + b
+    end
+
+    return -c/2 * ((t - 1) * (t - 3) - 1) + b
+end
+
+# Exponential Easing functions
+function EaseExpoIn(t::Real, b::Real, c::Real, d::Real)
+    return iszero(t) ? b : c * 2^(10(t/d - 1)) + b
+end
+function EaseExpoOut(t::Real, b::Real, c::Real, d::Real)
+    return iszero(t-d) ? b+c : c * (-(2.0^(-10 * t/d)) + 1) + b
```

Comment:
```suggestion
    return t ≈ d ? b+c : c * (-(2.0^(-10 * t/d)) + 1) + b
```

Generally this kind of check should use `isapprox` due to floating point precision.